File: tests/end-to-end/cloudwatch/typescript/stack.ts
import * as cdk from 'aws-cdk-lib';
import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';

export interface CloudwatchStackProps extends cdk.StackProps {
  /**
   * Environment used for this deployment.
   * @default 'dev'
   */
  readonly environmentName?: string;
}

export class CloudwatchStack extends cdk.Stack {
  public constructor(scope: cdk.App, id: string, props: CloudwatchStackProps = {}) {
    super(scope, id, props);

    // Applying default props
    props = {
      ...props,
      environmentName: props.environmentName ?? 'dev',
    };

    // Resources
    const myApi5xxErrorsAlarm = new cloudwatch.CfnAlarm(this, 'MyApi5xxErrorsAlarm', {
      alarmDescription: 'Example alarm',
      namespace: 'AWS/ApiGateway',
      dimensions: [
        {
          name: 'ApiName',
          value: 'MyApi',
        },
      ],
      metricName: '5XXError',
      comparisonOperator: 'GreaterThanThreshold',
      statistic: 'Average',
      threshold: 0,
      period: 900,
      evaluationPeriods: 1,
      treatMissingData: 'notBreaching',
      alarmActions: [
        cdk.Fn.importValue(`${props.environmentName!}AlarmsTopicArn`),
      ],
    });
  }
}

File: tests/end-to-end/cloudwatch/csharp/Stack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.CloudWatch;
using Constructs;
using System.Collections.Generic;

namespace CloudwatchStack
{
    public class CloudwatchStackProps : StackProps
    {
        /// <summary>
        /// Environment used for this deployment.
        /// </summary>
        public string EnvironmentName { get; set; }

    }

    public class CloudwatchStack : Stack
    {
        public CloudwatchStack(Construct scope, string id, CloudwatchStackProps props = null) : base(scope, id, props)
        {
            // Applying default props
            props ??= new CloudwatchStackProps();
            props.EnvironmentName ??= "dev";


            // Resources
            var myApi5xxErrorsAlarm = new CfnAlarm(this, "MyApi5xxErrorsAlarm", new CfnAlarmProps
            {
                AlarmDescription = "Example alarm",
                Namespace = "AWS/ApiGateway",
                Dimensions = new []
                {
                    new CfnAlarm.DimensionProperty
                    {
                        Name = "ApiName",
                        Value = "MyApi",
                    },
                },
                MetricName = "5XXError",
                ComparisonOperator = "GreaterThanThreshold",
                Statistic = "Average",
                Threshold = 0,
                Period = 900,
                EvaluationPeriods = 1,
                TreatMissingData = "notBreaching",
                AlarmActions = new []
                {
                    Fn.ImportValue($"{props.EnvironmentName}AlarmsTopicArn"),
                },
            });
        }
    }
}
